src/Polysemy/Scoped.hs

```diff
@@ -73,26 +71,28 @@ interpretScopedH' ::
     Tactical (Scoped param effect) (Sem r0) r x) ->
   InterpreterFor (Scoped param effect) r
 interpretScopedH' withResource scopedHandler =
-  go (errorWithoutStackTrace "top level run")
+  go 0 Empty
   where
-    go :: resource -> InterpreterFor (Scoped param effect) r
-    go resource =
+    go :: Word -> Seq resource -> InterpreterFor (Scoped param effect) r
+    go depth resources =
```

Comment:
This is how we can avoid `Opaque` pollution in the existing interpreters; we crop this pattern, of having a `Seq` (or `Vector`) of the allocated resources, and a `Run`'s depth is used as an index to acquire the correct resource.

That, however, means the existing interpreters can't be written in terms of `runScopedNew`. I'm honestly leaning towards applying this pattern to preserve backwards compatibility, but one day we _will_ scrap all these interpreters in favor of `runScopedNew`, and there's an argument to be made that we don't want to encourage people to keep using the old ones because they have _one_ advantage (no `Opaque` pollution) compared to `runScopedNew`.